blog/content/post/2020-04-27-trenchboot-nlnet-release-04.md

```diff
@@ -0,0 +1,290 @@
+abstract:
+cover: /covers/trenchboot-logo.png
+author: piotr.kleinschmidt
+layout: post
+published: false
+date: 2020-04-27
+archives: "2020"
+
+tags:
+  - trenchboot
+  - security
+  - open-source
+  - coreboot
+categories:
+  - Firmware
+  - Security
+
+---
+
+[Previous article](https://blog.3mdeb.com/2020/2020-04-03-trenchboot-nlnet-lz-validation/)
+showed how to enable DRTM on PC Engines apu2 platform and NixOS operating
+system. As project is constantly developed by us, there are regularly new
+features implemented. This article is showing how to update your DRTM and verify
+next project's requirements. Typically, new updates should be published every
+month as form of following blog posts and they will develop already configured
+system components. Therefore, **we strongly encourage to keep project up-to-date
+and keep track of TrenchBoot blog posts series**. If you haven't read previous
+articles, we recommend to catch up.
+
+## What's new
+
+We have made following changes since last release:
+
+- general **DRTM update in NixOS** (including landing-zone, Linux kernel and
+  Grub)
+- enabled DRTM in Yocto custom Linux built - **meta-trenchboot**
+- introduced **CI/CD system** to build each TrenchBoot components
+
+Each from above points is described with details later in this article. It is
+divided into sections:
+
+- Update DRTM in NixOS
+- Enable DRTM in custom Linux built
+- CI/CD system
+
+Depending on specific section, there are already met project's requirements
+mentioned and ways to verify them.
+
+## Update DRTM in NixOS
+
+Since last release, there are improvements in **landing-zone, grub and Linux
+kernel**. All of these components should be updated. Following procedure is
+showing how to do this properly.
+
+>Remember, that everything is done in NixOS and further verification is done for
+system precisely configured this way.
+
+      1. Pull `3mdeb/nixpkgs` repository.
+
+      ```
+      $ cd ~/nixpkgs
+      $ git branch trenchboot_support_2020.03
+      $ git pull
+      ```
+
+      2. Pull `3mdeb/nixos-trenchboot-configs` repository.
+
+      ```
+      $ cd ~/nixos-trenchboot-configs
+      $ git branch master
+      $ git pull
+      ```
+
+      3. Copy all configuration files to `/etc/nixos/` directory.
+
+      ```
+      $ cp nixos-trenchboot-configs/*.nix /etc/nixos
+      ```
+
+      4. Update system.
+
+      ```
+      $ sudo nixos-rebuild switch -I nixpkgs=~/nixpkgs
+      building Nix...
+      building the system configuration...
+      ```
+
+      5. Reboot platform
+
+      ```
+      $ reboot
+      ```
+
+
+
+#### Requirements verification - LZ content and layout
+
+There are two requirements which LZ must met:
+
+- bootloader information header in LZ should be moved at the end of the LZ
+binary or the end of code section of the LZ;
+- the size of the measured part of the SLB must be set to the code size only;
+
+Bootloader information header is special data structure which content is
```

Comment:
```suggestion
Bootloader information header is special data structure which layout is
```
Content is expected to be filled by a bootloader, so definitely not hardcoded

Reply:
Fixed